### PR TITLE
Chart labels, annotations, and priority class config

### PR DESCRIPTION
**Pull request by wozniakjan**
Kyma project currently uses fork of BTP Operator helm chart with few custom changes. We would like to upstream all configuration knobs and eventually deprecate the https://github.com/kyma-incubator/sap-btp-service-operator.

This is a first PR containing mainly just configurability for deployment annotations and priority class.

Previous PR https://github.com/SAP/sap-btp-service-operator/pull/201 has been closed without any review or comment and I am unable to reopen it.

### CODE REVIEW

**Comment by wozniakjan** [comment]
ptal @kerenlahav @pavelmaliy

File: sapbtp-operator-charts/templates/deployment.yml [hunk 1]
```diff
@@ -12,6 +12,8 @@ spec:
   selector:
     matchLabels:
       control-plane: controller-manager
+      app.kubernetes.io/instance: sap-btp-operator
+      app.kubernetes.io/name: sap-btp-operator
```
Comment (kerenlahav) [inline sapbtp-operator-charts/templates/deployment.yml:16]: What are these labels used for?

Comment (wozniakjan) [reply sapbtp-operator-charts/templates/deployment.yml:16]: thanks for catching this! my apologies, this should be templated.

Comment (wozniakjan) [reply sapbtp-operator-charts/templates/deployment.yml:16]: these already exist as labels for the pod
https://github.com/SAP/sap-btp-service-operator/blob/e166d789a6b6f10fe71e8ce11281a95e89d2dae1/sapbtp-operator-charts/templates/deployment.yml#L30-L31

there was just discrepancy between `labels` and `matchLabels`

File: sapbtp-operator-charts/templates/deployment.yml [hunk 2]
```diff
@@ -20,6 +22,9 @@ spec:
         {{- $secretTls := (include (print $.Template.BasePath "/secret-tls.yml") .) -}}
         {{- $configSha := (print $configmap $secret $secretTls) | sha256sum }}
         checksum/config: {{ $configSha }}
+        {{- if .Values.manager.annotations }}
+        {{- toYaml .Values.manager.annotations | nindent 8 }}
```
Comment (kerenlahav) [inline sapbtp-operator-charts/templates/deployment.yml:26]: how is this used? should a new entry be added to the values.yaml?

File: sapbtp-operator-charts/templates/deployment.yml [hunk 3]
```diff
@@ -112,6 +117,9 @@ spec:
       imagePullSecrets: {{ toYaml .Values.manager.imagePullSecrets | nindent 8 }}
     {{- end }}
       terminationGracePeriodSeconds: 10
+      {{- if .Values.manager.priorityClassName }}
+      priorityClassName: {{ .Values.manager.priorityClassName }}
```
Comment (kerenlahav) [inline sapbtp-operator-charts/templates/deployment.yml:121]: same as last comment

File: sapbtp-operator-charts/templates/deployment.yml (hunk 2 above)
Comment (wozniakjan) [reply sapbtp-operator-charts/templates/deployment.yml:26]: this is just allowing consumers of this chart to provide custom annotations. In kyma, we use this to disable isto injection.

**Review by kerenlahav — APPROVED** [review]

**Pull request merged**